src/components/common/img/Img.js

```diff
@@ -82,18 +91,45 @@ class Img extends Component {
         this.displayImage.src = image
     }
 
+    renderFullscreenIconAndDialog() {
+        const { classes, allowFullscreen } = this.props
+        if (!allowFullscreen) {
+            return null
+        }
+        return (
+            <div>
+                <IconButton
+                    className={classnames(classes.fullscreenIcon)}
+                    onClick={() => { this.handleDialogToggle() }}
+                >
+                    <FullscreenIcon/>
+                </IconButton>
+                <FullscreenImgDialog
```

Comment:
consider FullscreenImgDialog active state to logic and  IconButton to be inside it (like WithDialog hoc).
It will be easy if it is used from other component. Otherwise it can be just render function here in Img component. In that case you can escape loading the Img again as it is used in FullscreenImgDialog